### PR TITLE
Reimplement location examples using component

### PR DESCRIPTION
**Pull request by LukasPaczos**
Reimplements all location examples with `LocationComponent` and removes `LocationLayerPlugin` dependency.

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java [hunk 1]
```diff
@@ -1,29 +1,32 @@
 package com.mapbox.mapboxandroiddemo.examples.plugins;
 
-// #-code-snippet: location-plugin-activity full-java
+// #-code-snippet: location-component-activity full-java
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java]: @langsmith does this change need to be reflected in the `android-docs` repo?

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java [hunk 2]
```diff
@@ -132,4 +145,4 @@ public void onLowMemory() {
     mapView.onLowMemory();
   }
 }
-// #-end-code-snippet: location-plugin-activity full-java
\ No newline at end of file
+// #-end-code-snippet: location-component-activity full-java
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java]: same as above

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java [hunk 3]
```diff
@@ -13,25 +13,19 @@
...
-import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
 import com.mapbox.mapboxsdk.maps.SupportMapFragment;
-import com.mapbox.mapboxsdk.plugins.locationlayer.LocationLayerPlugin;
-import com.mapbox.mapboxsdk.plugins.locationlayer.modes.CameraMode;
-import com.mapbox.mapboxsdk.plugins.locationlayer.modes.RenderMode;
 
 import java.util.List;
 
-// #-code-snippet: location-plugin-fragment-activity full-java
+// #-code-snippet: location-component-fragment-activity full-java
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java]: same as above

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java [hunk 4]
```diff
@@ -112,22 +98,14 @@ public void onExplanationNeeded(List<String> permissionsToExplain) {
...
+      enableLocationComponent();
     } else {
       Toast.makeText(this, R.string.user_location_permission_not_granted, Toast.LENGTH_LONG).show();
       finish();
     }
   }
 }
-// #-end-code-snippet: location-plugin-fragment-activity full-java
\ No newline at end of file
+// #-end-code-snippet: location-component-fragment-activity full-java
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java]: same as above

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java]: you can actually remove that `code-snippet` line from the top and bottom of the file. https://github.com/mapbox/mapbox-android-demo/pull/870 got rid of them for all other examples

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java]: see https://github.com/mapbox/mapbox-android-demo/pull/878#discussion_r223681920

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java (hunk 3 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java]: see https://github.com/mapbox/mapbox-android-demo/pull/878#discussion_r223681920

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java (hunk 4 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentFragmentActivity.java]: see https://github.com/mapbox/mapbox-android-demo/pull/878#discussion_r223681920

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationComponentActivity.java]: and no, no changes are needed in `/android-docs`

**Comment by LukasPaczos** [comment]
This one is ready for a review @langsmith

**Comment by LukasPaczos** [comment]
https://github.com/mapbox/mapbox-android-demo/pull/878/commits/513413797152fe294001fb998800a16e66cc76bb adds a new `Location` category to the drawer and I moved `LocationComponentFragmentActivity`, `LocationComponentActivity` and `KotlinLocationComponentActivity` in there.

**Pull request merged**